src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Services/Helpers/DateTimeProviderContext.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections;
+using System.Threading;
+
+namespace DevAdventCalendarCompetition.Services
+{
+    public class DateTimeProviderContext : IDisposable
+    {
+        private static readonly ThreadLocal<Stack> ThreadScopeStack = new ThreadLocal<Stack>(() => new Stack());
+
+        public DateTimeProviderContext(DateTime contextDateTimeUtcNow)
+        {
+            this.ContextDateTimeNow = contextDateTimeUtcNow;
+            ThreadScopeStack.Value.Push(this);
+        }
+
+        public static DateTimeProviderContext Current
+        {
+            get
+            {
+                if (ThreadScopeStack.Value.Count == 0)
+                {
+                    return null;
```

Comment:
See the warning - Added line #L23 was not covered by tests

Reply:
Actually, I don't see it, but I see some disadvantages to this solution. It is called the "Ambient context anti-pattern" for a reason. I'm may propose another solution. Different class for checking IsAdvent logic that has injected AdventSettings and IDateTime.